src/InstrumentationEngine.Attach/Program.cs

```diff
@@ -136,26 +235,88 @@ internal static int Main(string[] args)
                 return ExitCodeFailure;
             }
 
+            #endregion
+
+            #region Profiler Attach
             // CONSIDER: Should the engine be validated before attempting to attach it to the CLR?
             // For example, could check the signature on the assembly (which wouldn't work on Linux).
 
             DiagnosticsClient client = new DiagnosticsClient(processId);
             try
             {
-                client.AttachProfiler(TimeSpan.FromSeconds(10), InstrumentationEngineClsid, enginePath);
+                client.AttachProfiler(TimeSpan.FromSeconds(10), InstrumentationEngineClsid, enginePath, bytes);
             }
             catch (ServerErrorException ex)
             {
                 WriteError(Invariant($"Could not attach engine to process: '{ex.Message}'."));
                 return ExitCodeFailure;
             }
 
+            #endregion
+
             return ExitCodeSuccess;
         }
 
         private static void WriteError(string message)
         {
             Console.Error.WriteLine(message);
         }
+
+        private static bool TryParseEngineConfiguration(
+            InstrumentationConfigurationSources sources,
+            Chip targetChip,
+            out InstrumentationEngineConfiguration configuration)
+        {
+            configuration = new InstrumentationEngineConfiguration();
+            configuration.InstrumentationEngine = new InstrumentationEngine()
```

Comment:
>InstrumentationEngine [](start = 54, length = 21)

I'm currently hardcoding the engine settings, but they can be parsed from a configuration xml as well.

Reply:
Or cmdline parameters

---
In reply to: [387369739](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r387369739) [](ancestors = 387369739)